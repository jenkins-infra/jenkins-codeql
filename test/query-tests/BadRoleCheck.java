import hudson.remoting.Callable;
import org.jenkinsci.remoting.Role;
import org.jenkinsci.remoting.RoleChecker;
import org.jenkinsci.remoting.RoleSensitive;
import jenkins.security.MasterToSlaveCallable;

public class BadRoleCheck {
    public static class OkCallable implements Callable {
        public void checkRoles(RoleChecker checker) throws SecurityException {
            checker.check(this, new Role("slave"));
        }
    }

    public static class StubCallable implements Callable {
        public void checkRoles(RoleChecker checker) throws SecurityException {
            // stub
        }
    }

    public static class WrongCallable implements Callable {
        public void checkRoles(RoleChecker checker) throws SecurityException {
            System.err.println("Hello, world!");
        }
    }

    public static class InheritedOkCallable extends MasterToSlaveCallable {
    }

    public static class InheritedOverrideCallable extends MasterToSlaveCallable {
        public void checkRoles(RoleChecker checker) throws SecurityException {
            // stub
        }
    }
}
