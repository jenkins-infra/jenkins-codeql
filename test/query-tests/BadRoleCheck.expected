| BadRoleCheck.java:15:21:15:30 | checkRoles | Potentially unsafe Callable implementation StubCallable#checkRoles |
| BadRoleCheck.java:21:21:21:30 | checkRoles | Potentially unsafe Callable implementation WrongCallable#checkRoles |
| BadRoleCheck.java:30:21:30:30 | checkRoles | Potentially unsafe Callable implementation InheritedOverrideCallable#checkRoles |
